web/modules/custom/joinup_discussion/src/EventSubscriber/SubscribedDiscussionSubscriber.php

```diff
@@ -57,75 +69,191 @@ class SubscribedDiscussionSubscriber implements EventSubscriberInterface {
+   *
+   * @return array
+   *   An associative array of message arguments, keyed by argument ID.
+   *
+   * @throws \Exception
+   *   Thrown if one of the arguments could not be generated.
    */
-  protected function getArguments(): array {
-    return [
-      '@entity:title' => $this->discussion->label(),
-      '@group:label' => $this->group->label(),
-      '@group:bundle' => $this->group->bundle(),
-      '@entity:url' => $this->discussion->toUrl('canonical', [
-        'absolute' => TRUE,
-      ])->toString(),
-    ];
+  protected function getArguments(NodeInterface $discussion): array {
+    $arguments = [];
+
+    $arguments['@entity:title'] = $discussion->label();
+    try {
+      $entity_url = $discussion->toUrl('canonical', ['absolute' => TRUE])->toString();
+    }
+    catch (EntityMalformedException | UndefinedLinkTemplateException $e) {
+      // Retrieval of the URL might fail if the link template is not defined, or
+      // if the entity doesn't have an ID.
+      throw new \Exception('Could not retrieve URL for discussion.', NULL, $e);
+    }
+    $arguments['@entity:url'] = $entity_url;
+
+    $actor = $this->getCurrentUser();
+    $actor_first_name = !empty($actor->get('field_user_first_name')->value) ? $actor->get('field_user_first_name')->value : '';
+    $actor_family_name = !empty($actor->get('field_user_family_name')->value) ? $actor->get('field_user_family_name')->value : '';
+
+    if ($actor->hasRole('moderator')) {
+      $arguments['@actor:full_name'] = 'The Joinup Support Team';
+    }
+    else {
+      $arguments['@actor:full_name'] = empty($actor->get('full_name')->value) ? $actor_first_name . ' ' . $actor_family_name : $actor->get('full_name')->value;
+    }
+
+    $group = $this->getDiscussionGroup($discussion);
+    if ($group) {
+      $arguments['@group:label'] = $group->label();
+      $arguments['@group:bundle'] = $group->bundle();
+    }
+
+    return $arguments;
+  }
+
+  /**
+   * Returns the fully loaded User entity for the current user.
+   *
+   * @return \Drupal\user\UserInterface
+   *   The current user.
+   */
+  protected function getCurrentUser(): UserInterface {
+    try {
+      /** @var \Drupal\user\UserInterface $user */
+      $user = $this->entityTypeManager->getStorage('user')->load($this->currentUser->id());
+      return $user;
+    }
+    catch (InvalidPluginDefinitionException $e) {
+      // The storage for the 'user' entity type should exist. If it didn't
+      // Drupal would be completely broken. This code should never run but we
+      // are returning an anonymos user entity for completeness and to satisfy
+      // the inspections of the IDE.
+      return new User([], 'user');
+    }
+  }
+
+  /**
+   * Sends the notification to the recipients.
+   *
+   * @param \Drupal\node\NodeInterface $discussion
+   *   The discussion for which to send the notification.
+   * @param string $message_template
+   *   The ID of the message template to use.
+   *
+   * @return bool
```

Comment:
In fact the caller never uses this return value. I'm not sure we need to return something here. Also, even Drupal send successfully an email, there's no guarantee the the message has been delivered by the system mailing and we cannot do anything in case of failure. I would make this function `:void` and just catch and log a potential exception (?)